pal/src/map/common.cpp

```diff
@@ -49,7 +49,7 @@ INT W32toUnixAccessControl( IN DWORD flProtect )
         MemAccessControl = PROT_EXEC | PROT_READ | PROT_WRITE;
         break;
     case PAGE_EXECUTE :
-        MemAccessControl = PROT_EXEC;
+        MemAccessControl = PROT_EXEC | PROT_READ; // WinAPI PAGE_EXECUTE also implies readable
```

Comment:
Try this:

    #include "windows.h"
    #include "stdio.h"
    void main()
    {

        void * address = VirtualAlloc(NULL, 64 * 1024, MEM_COMMIT | MEM_RESERVE, PAGE_EXECUTE);
        int i;
        SIZE_T readCount;
        if (!ReadProcessMemory(GetCurrentProcess(), address, &i, sizeof(i), &readCount))
        {
            printf("Unable to read\n");
        }
        else
        {
            printf("Read %d, count %d\n", i, readCount);
        }
    }

I got
`Read 0, count 4`

I am not sure you can distinguish different reads.  If you can read, you can read, and it seems like on windows PAGE_EXECUTE grants you read permission?

I don't mind us switching to PAGE_EXECUTE_READ to be more explicit as well.  But the PAL should emulate what windows API does.